src/PointwiseFunctions/Hydro/InitialData/IrrotationalBns.hpp

```diff
@@ -0,0 +1,126 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/Gsl.hpp"
+
+/*!
+ * \ingroup HydroInitialDataGroup
+ * \brief Items related to solving for irrotational bns initial data
+ * See e.g. \cite BaumgarteShapiro Ch. 15 (P. 523)
+ */
+namespace hydro::initial_data::irrotational_bns {
+/// @{
+/// \brief Compute the  shift plus a spatial vector \f$ k^i\f$ representing
+/// the local binary rotation \f$B^i = \beta^i + k^i\f$
+///
+template <typename DataType>
+void rotational_shift(
+    gsl::not_null<tnsr::I<DataType, 3>*> result,
+    const tnsr::I<DataType, 3>& shift,
+    const tnsr::I<DataType, 3>& spatial_rotational_killing_vector);
+
+template <typename DataType>
+tnsr::I<DataType, 3> rotational_shift(
+    const tnsr::I<DataType, 3>& shift,
+    const tnsr::I<DataType, 3>& spatial_rotational_killing_vector);
+/// @}
+
+/// @{
+/// \brief Compute the  stress-energy corresponding to the rotation shift
+///
+///
+/// \f[\Sigma^i_j = \frac{1}{2}\frac{B^iB_j}{\alpha^2}\f]
+///
+template <typename DataType>
+void rotational_shift_stress(gsl::not_null<tnsr::II<DataType, 3>*> result,
+                             const tnsr::I<DataType, 3>& rotational_shift,
+                             const Scalar<DataType>& lapse);
+template <typename DataType>
+tnsr::II<DataType, 3> rotational_shift_stress(
+    const tnsr::I<DataType, 3>& rotational_shift,
+    const Scalar<DataType>& lapse);
+/// @}
+
+/// @{
+/// \brief  Compute derivative  \f$D_i B^j / \alpha \f$
+///
+/// Here D_i is the spatial covariant derivative, \f$ \alpha\f$ is the lapse
+/// and \f$B^i\f$ the rotational shift)
```

Comment:
Please clarify that you want the _covariant_ spatial derivatives as input here, not the partial derivatives, if I understand correctly. Is it indeed easier to add Christoffel symbols elsewhere, or would it make sense to add them here?

Related question: would it be helpful to implement some of these functions in terms of the conformal metric and the conformal factor, rather than the spatial metric? In a combined XCTS+phi system we'll have the conformal metric with its Christoffel symbols readily available. Of course we can always compute the spatial metric, and also the spatial Christoffel symbols with Eq. (3.7) in Baumgarte/Shapiro. Whichever is easier.

Reply:
The way it was written  was confusing, I think it makes sense for this to just compute the partial derivatives and worry about the Christoffels later.  The cases where this is actually used I think may not need to compute full Christoffels, just the contracted ones.  

I think in general it makes sense to keep things as general as possible (i.e. not specializing to the XCTS system).  We may not always have an XCTS solve available.